tests/integration/test_network_configuration.py

```diff
@@ -0,0 +1,248 @@
+# host that services make requests to when starting up (e.g. opensearch) and
+# they won't start if we override the variable.
+
+
+pytestmark = [pytest.mark.only_localstack]
+
+
+class TestOpenSearch:
+    """
+    OpenSearch does not respect any customisations and just returns a domain with localhost.localstack.cloud in.
+    """
+
+    def test_default_strategy(
+        self, opensearch_client, opensearch_wait_for_cluster, assert_host_customisation
+    ):
+        domain_name = f"domain-{short_uid()}"
+        res = opensearch_client.create_domain(DomainName=domain_name)
+        opensearch_wait_for_cluster(domain_name)
+        endpoint = res["DomainStatus"]["Endpoint"]
+
+        assert_host_customisation(endpoint, use_localstack_cloud=True)
+
+    def test_port_strategy(
+        self, monkeypatch, opensearch_client, opensearch_wait_for_cluster, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "OPENSEARCH_ENDPOINT_STRATEGY", "port")
+
+        domain_name = f"domain-{short_uid()}"
+        res = opensearch_client.create_domain(DomainName=domain_name)
+        opensearch_wait_for_cluster(domain_name)
+        endpoint = res["DomainStatus"]["Endpoint"]
+
+        if config.is_in_docker:
+            assert_host_customisation(endpoint, use_localhost=True)
+        else:
+            assert_host_customisation(endpoint, custom_host="127.0.0.1")
+
+    def test_path_strategy(
+        self, monkeypatch, opensearch_client, opensearch_wait_for_cluster, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "OPENSEARCH_ENDPOINT_STRATEGY", "path")
+
+        domain_name = f"domain-{short_uid()}"
+        res = opensearch_client.create_domain(DomainName=domain_name)
+        opensearch_wait_for_cluster(domain_name)
+        endpoint = res["DomainStatus"]["Endpoint"]
+
+        assert_host_customisation(endpoint, use_localstack_hostname=True)
+
+
+class TestS3:
+    @pytest.mark.skipif(
+        condition=config.LEGACY_S3_PROVIDER, reason="Not implemented for legacy provider"
+    )
+    def test_non_us_east_1_location(
+        self, monkeypatch, s3_resource, s3_client, cleanups, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "LEGACY_S3_PROVIDER", False)
+        monkeypatch.setenv("PROVIDER_OVERRIDE_S3", "asf")
+
+        bucket_name = f"bucket-{short_uid()}"
+        res = s3_client.create_bucket(
+            Bucket=bucket_name,
+            CreateBucketConfiguration={
+                "LocationConstraint": "eu-west-1",
+            },
+        )
+
+        def cleanup():
+            bucket = s3_resource.Bucket(bucket_name)
+            bucket.objects.all().delete()
+            bucket.object_versions.all().delete()
+            bucket.delete()
+
+        cleanups.append(cleanup)
+
+        assert_host_customisation(res["Location"], use_hostname_external=True)
+
+    def test_multipart_upload(self, s3_bucket, s3_client, assert_host_customisation):
```

Comment:
This multipart test made me think on where the `Location` was updated for these responses, and it seems like there's still a use of `HOSTNAME_EXTERNAL` in `s3_listener.py` here: https://github.com/localstack/localstack/blob/4914f4896e86df94e7b815c89815257b6a73c262/localstack/services/s3/s3_listener.py#L1366

Not sure if this needs to be update as well?

Reply:
Thanks for pointing this out - yeah that did need an update as well. I've pushed a new commit.